src/server/snap.rs

```diff
@@ -86,17 +85,15 @@ fn send_snap(mgr: SnapManager, addr: SocketAddr, data: ConnData) -> Result<()> {
     let snap = data.msg.get_raft().get_message().get_snapshot();
     let key = try!(SnapKey::from_snap(&snap));
     mgr.wl().register(key.clone(), SnapEntry::Sending);
-    let snap_file = box_try!(mgr.rl().get_snap_file(&key, true));
+    let mut f = box_try!(mgr.rl().get_send_snapshot_file(&key));
     defer!({
-        snap_file.delete();
```

Comment:
It's fine to leave the snapshot file if any error occurs when it's being sent out, since it will be cleaned up eventually by GC.